tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -2652,17 +2654,26 @@ LogicalResult ConvertTFLSliceOp::matchAndRewrite(
     return failure();
   }
 
-  for (int i = 0; i < begin_elems.getNumElements(); i++)
-    begin_vals.push_back(begin_elems.getValues<APInt>()[i].getSExtValue());
+  if (input_type.hasRank()) {
+    ArrayRef<int64_t> shape = input_type.getShape();
+    size_t rank = shape.size();
+    assert(begin_elems.getNumElements() == rank);
+    assert(size_elems.getNumElements() == rank);
 
-  for (int i = 0; i < size_elems.getNumElements(); i++)
-    size_vals.push_back(size_elems.getValues<APInt>()[i].getSExtValue());
+    for (int i = 0; i < rank; i++) {
+      int64_t begin = begin_elems.getValues<APInt>()[i].getSExtValue();
+      int64_t size = size_elems.getValues<APInt>()[i].getSExtValue();
+      size = (size == -1) ? shape[i] - begin : size;
```

Comment:
Is there a constant that could be used instead? -1 is tricky to find again/many different uses as sentinel elsewhere.

Reply:
I'd have ideally preferred one in TFLite already. But yes, just create a const local variable here so that at least the -1 has a name.